Make SetMiscWarning() accept bilingual_str argument

hebasto opened the pull request:
This is one more step for consistent usage of `bilingual_str`.

No new translation messages are defined.

hebasto committed:
refactor: Make GetWarnings() bilingual_str aware internally

hebasto committed:
Make GetWarnings() return bilingual_str

hebasto committed:
Make SetMiscWarning() accept bilingual_str argument

MarcoFalke commented on src/warnings.cpp:
```diff
@@ -6,18 +6,21 @@
-static std::string strMiscWarning GUARDED_BY(g_warnings_mutex);
+#include <vector>
+
+static Mutex g_warnings_mutex;;
```
```suggestion
static Mutex g_warnings_mutex;
```

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/19233#issuecomment-641903384).

hebasto commented:
Updated 45cb61a835711df350ebee6a8ebddb8dc23ab47d -> ddafc58af052741d916a242c4a7df16e468efbdd ([pr19233.01](https://github.com/hebasto/bitcoin/commits/pr19233.01) -> [pr19233.02](https://github.com/hebasto/bitcoin/commits/pr19233.02), [diff](https://github.com/hebasto/bitcoin/compare/pr19233.01..pr19233.02)):

- removed [extra]((https://github.com/bitcoin/bitcoin/pull/19233#discussion_r438008592)) `;`

MarcoFalke commented on src/warnings.cpp:
```diff
@@ -38,34 +41,36 @@ void SetfLargeWorkInvalidChainFound(bool flag)
     if (!CLIENT_VERSION_IS_RELEASE) {
-        warnings_concise = "This is a pre-release test build - use at your own risk - do not use for mining or merchant applications";
-        warnings_verbose = _("This is a pre-release test build - use at your own risk - do not use for mining or merchant applications").translated;
+        warnings_concise = _("This is a pre-release test build - use at your own risk - do not use for mining or merchant applications.");
```
why change the translation?
